Dockerfile

```diff
@@ -1,46 +1,95 @@
-FROM        registry.access.redhat.com/ubi9/python-39:9.5-1739799514@sha256:9bbc4cfeac896544ab3eafa088c3d6995e82592362d374606f00d221f2986fe0 as prod
+##############
+# base image #
+##############
+FROM registry.access.redhat.com/ubi9/python-39:9.5-1739799514@sha256:9bbc4cfeac896544ab3eafa088c3d6995e82592362d374606f00d221f2986fe0 as base
 
+COPY        LICENSE /licenses/LICENSE
+
+#################
+# builder image #
+#################
+FROM base as builder
+
+# Get the uv binary from upstream
 COPY --from=ghcr.io/astral-sh/uv:0.6.11@sha256:fb91e82e8643382d5bce074ba0d167677d678faff4bd518dac670476d19b159c /uv /bin/uv
 
-COPY --chown=1001:0 pyproject.toml uv.lock README.md /dashdotdb/
-COPY        LICENSE /licenses/LICENSE
+ENV \
+  UV_COMPILE_BYTECODE="true" \
+  UV_NO_CACHE=true \
+  # inherits from base image
+  UV_PROJECT_ENVIRONMENT=$APP_ROOT
+
+COPY --chown=1001:0 \
+    pyproject.toml  \
+    uv.lock         \
+  /dashdotdb/
+
 USER        1001
 WORKDIR     /dashdotdb
```

Comment:
these 2 are not need, as current user in base is `1001` already, and we can use default workdir `/opt/app-root/src`